### PR TITLE
add sparse support for sum op

### LINKED ISSUE #5094: need to refine sum operator for sparse update
In sparse update, the gradients can be SelectedRows. We have to sum these sparse gradients first, and then use sgd operator to update the parameter. So, sum operator must support SelectedRows sum to SelectedRow and SelectedRows sum to Tensor in backward pass.

### PR DESCRIPTION
**Pull request by QiJune**
Fix #5094 
This PR depends on #5027 
Because current operator backward tests are still based on run-time, not compile-time, we can not set correct Variable type in backward process.
I comment check_grad of test_sum_op for an example. test_word2vec and test_sum_op can pass ci test.

### CODE REVIEW

File: paddle/framework/executor.cc [hunk 1]
```diff
@@ -56,6 +57,22 @@ Executor::~Executor() {
   }
 }
 
+void CreateTensor(Variable* var, VarDesc::VarType var_type) {
```
Comment (reyoung) [inline paddle/framework/executor.cc]: `static void CreateTensor()`

File: paddle/framework/selected_rows.h [hunk 2]
```diff
@@ -37,6 +37,8 @@ class SelectedRows {
 
   const Vector<int64_t>& rows() const { return rows_; }
 
+  Vector<int64_t>& mutable_rows() { return rows_; }
```
Comment (reyoung) [inline paddle/framework/selected_rows.h]: `Vector<int64_t>* mutable_rows()`

File: paddle/operators/math/selected_rows_functor.cc [hunk 3]
```diff
@@ -109,6 +109,69 @@ struct SelectedRowsAddTensor<platform::CPUPlace, T> {
...
+    auto* in1_data = in1_value.data<T>();
+    auto* in2_data = in2_value->data<T>();
+    memory::Copy(boost::get<platform::CPUPlace>(in2_place),
+                 in2_data + input2_offset,
+                 boost::get<platform::CPUPlace>(in1_place), in1_data,
+                 in1_value.numel() * sizeof(T));
+  }
+};
+
+template struct SelectedRowsAddTo<platform::CPUPlace, float>;
```
Comment (reyoung) [inline paddle/operators/math/selected_rows_functor.cc]: add double

File: paddle/framework/executor.cc (hunk 1 above)
Comment (QiJune) [reply paddle/framework/executor.cc]: Done

File: paddle/framework/selected_rows.h (hunk 2 above)
Comment (QiJune) [reply paddle/framework/selected_rows.h]: Done

File: paddle/operators/math/selected_rows_functor.cc (hunk 3 above)
Comment (QiJune) [reply paddle/operators/math/selected_rows_functor.cc]: Done

File: paddle/operators/sum_op.h [hunk 4]
```diff
@@ -25,19 +29,68 @@ template <typename Place, typename T>
...
+
+    if (out_var->IsType<framework::LoDTensor>()) {
+      auto* out = context.Output<Tensor>("Out");
+      // Runtime InferShape
+      for (int i = 0; i < N; i++) {
+        if (in_vars[i]->IsType<framework::LoDTensor>()) {
+          out->Resize(in_vars[i]->Get<framework::LoDTensor>().dims());
+          break;
+        }
+      }
```
Comment (reyoung) [inline paddle/operators/sum_op.h]: 39-44 lines are not necessary

**Pull request merged**